### DISCUSSION TITLE
Get All Events Message - BaseEventType

### DISCUSSION THREAD

**Discussion by rafaelmoraes33** [FreeOpcUa/python-opcua#1536] [General]
Hello, I'm trying to find some way to read values from the BaseEventType variable type.

I have to create a routine where every minute I check if there is a new event and get the columns, Message, Time and Severity.

With normal variable type I can do it perfectly.

I would greatly appreciate the help.

Thanks

![Untitled](https://github.com/FreeOpcUa/python-opcua/assets/45277330/cf1d91a6-1cfe-4f3b-97e2-417cebc1b0bb)

**Comment by schroeder-** [comment]
You need to use subscription of events. See https://github.com/FreeOpcUa/python-opcua/blob/master/examples/client-events.py

**Reply by rafaelmoraes33** [reply]
Hi!!

<img width="1559" alt="Screen Shot 2024-01-22 at 10 51 09 AM" src="https://github.com/FreeOpcUa/python-opcua/assets/45277330/581e518e-4d5d-4b65-a276-9305cbf4d357">

I'm using this example, but it's giving this error.

Would you help me?

Thanks

**Reply by schroeder-** [reply]
I condenst the example a bit more.... :

```

class SubHandler(object):

    """
    Subscription Handler. To receive events from server for a subscription
    data_change and event methods are called directly from receiving thread.
    Do not do expensive, slow or network operation there. Create another
    thread if you need to do such a thing
    """
    def event_notification(self, event):
        print("New event recived: ", event)


if __name__ == "__main__":

    client = Client("opc.tcp://localhost:4840/freeopcua/server/")
    # client = Client("opc.tcp://admin@localhost:4840/freeopcua/server/") #connect using a user
    try:
        client.connect()
        msclt = SubHandler()
        sub = client.create_subscription(100, msclt)
        handle = sub.subscribe_events()
        embed()
        sub.unsubscribe(handle)
        sub.delete()
    finally:
        client.disconnect() 
```
